lib/nokogiri/xml/node.rb

```diff
@@ -192,16 +198,47 @@ def decorate!
 
       # :section: Manipulating Document Structure
 
-      ###
-      # Add +node_or_tags+ as a child of this Node.
-      #
-      # +node_or_tags+ can be a Nokogiri::XML::Node, a ::DocumentFragment, a ::NodeSet, or a String
-      # containing markup.
-      #
-      # Returns the reparented node (if +node_or_tags+ is a Node), or NodeSet (if +node_or_tags+ is
-      # a DocumentFragment, NodeSet, or String).
-      #
-      # Also see related method +<<+.
+      # :call-seq:
+      #   add_child(object) -> object
+      #
+      # Appends one or more children (as specified by the given +object+) to +self+;
+      # returns +object+.
+      #
+      # When +object+ is an XML::Node of any kind,
+      # adds the node as the last child of +self+;
+      # sets the parent of that node as +self+:
+      #
+      #   doc = XML::Document.new
+      #   doc.children.size      # => 0
+      #   root_ele = XML::Element.new('root', doc)
+      #   root_ele.parent        # => nil
+      #   doc.add_child(root_ele)
+      #   doc.children.size      # => 1
+      #   root_ele.parent == doc # => true
+      #
+      # When +object+ is an XML::NodeSet,
+      # appends each node in the nodeset to the children of +self+, as above:
+      #
+      #   other_doc = XML::Document.new
+      #   root_ele = XML::Element.new('other', other_doc)
+      #   other_doc.add_child(root_ele)
+      #   # Move all book nodes from bookstore_doc to other_doc.root.
+      #   bookstore_doc = fresh_bookstore_doc
+      #   book_nodes = bookstore_doc.search('//book')
+      #   other_doc.root.add_child(book_nodes)
+      #   bookstore_doc.search('//book').size # => 0
+      #   other_doc.search('//book').size     # => 4
+      #
+      # When +object+ is a string,
+      # creates a new XML::Node from the string
+      # and adds the node as the last child of +self+:
```

Comment:
I don't think this illustrates the feature very well, which is intended to parse a string into a node (or nodes) and add them. How about something like:

```ruby
doc = XML::Document.parse("<root></root>")
doc.root.add_child("<child>first</child><child>second</child>")
doc.root.children.size # => 2
```